js/handlers/create-post.mjs

```
@@ -0,0 +1,29 @@
+import { createPost } from "../api/posts/createPost.mjs";
+import { displayProfile } from "../render/profile.mjs";
+
+//handle create post form submission
+export const createPostForm = document.getElementById("create-post-form");
+
+export function handleCreatePostForm() {
+  if (!createPostForm) {
+    return;}
+
+    createPostForm.addEventListener("submit",async (event) => {
+      event.preventDefault();
+
+      const post = {
+        title: createPostForm.title.value,
+        body: createPostForm.body.value,
+        media: createPostForm.media.value,
+      };
+
+    // send to API and wait for the operation to complete
+    try {
+      await createPost(post);
+      window.location.reload();  // reload the page to show the new post
+  } catch (error) {
+      throw new Error(error.message);
+  }
+      
```

`catch block`: to suppress an error
`throw`: to create an error

1) Understand the differences of the above, they have different purposes.

2) Formating and indentation.